src/passes/ifStatementTempVarPostpender.ts

```diff
@@ -0,0 +1,192 @@
+    // analyse the block
+    if (node.vBody) {
+      this.dispatchVisit(node.vBody, ast);
+    }
+    // remove the parameters and implicits
+    [...node.implicits].forEach((implicit) => {
+      if (!this.liveVars.delete(implicit)) {
+        throw new TranspileFailedError(
+          `Implicit variable dead. ${implicit} in function ${node.name}.`,
+        );
+      }
+    });
+    this.liveVars = new Set<string>();
+  }
+
+  visitIdentifier(node: Identifier, _ast: AST): void {
+    if (
+      node.vReferencedDeclaration !== undefined &&
+      node.vReferencedDeclaration.getClosestParentByType(CairoFunctionDefinition) !== undefined
+    ) {
+      this.liveVars.add(node.name);
+    }
+  }
+
+  visitBlock(node: Block, ast: AST): void {
+    node.vStatements
+      .slice(0)
+      .reverse()
+      .map((child) => this.dispatchVisit(child, ast));
+  }
+
+  visitUncheckedBlock(node: UncheckedBlock, ast: AST): void {
+    node.vStatements
+      .slice(0)
+      .reverse()
+      .map((child) => this.dispatchVisit(child, ast));
+  }
+
+  visitVariableDeclarationStatement(node: VariableDeclarationStatement, ast: AST): void {
+    node.vDeclarations.forEach((declaration) => this.liveVars.delete(declaration.name));
+    if (node.vInitialValue) {
+      this.dispatchVisit(node.vInitialValue, ast);
+    }
+  }
+
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    // ignore the function identifier
+    for (const arg of node.vArguments) {
+      this.dispatchVisit(arg, ast);
+    }
+  }
+
+  visitIfStatement(node: IfStatement, ast: AST): void {
+    ensureBothBranchesAreBlocks(node, ast);
+    assert(node.vFalseBody instanceof Block);
+
+    const liveVarsBefore = new Set(this.liveVars);
+    this.dispatchVisit(node.vTrueBody, ast);
+    const liveVarsAfterTrueBody = new Set(this.liveVars);
+
+    this.liveVars = new Set(liveVarsBefore);
+    let liveVarsAfterFalseBody = new Set<string>();
+
+    this.dispatchVisit(node.vFalseBody, ast);
+    liveVarsAfterFalseBody = new Set(this.liveVars);
+
+    this.liveVars = new Set(liveVarsBefore);
+    this.dispatchVisit(node.vCondition, ast);
+
+    this.liveVars = new Set([
+      ...this.liveVars,
+      ...liveVarsAfterFalseBody,
+      ...liveVarsAfterTrueBody,
+    ]);
+
+    const trueHasPathWithoutReturn = hasPathWithoutReturn(node.vTrueBody);
+    if (trueHasPathWithoutReturn) {
+      for (const liveVar of liveVarsBefore) {
+        const child = createCairoTempVar(liveVar, ast);
+        (node.vTrueBody as Block).appendChild(child);
+        ast.registerChild(child, node.vTrueBody);
+      }
+    }
+    const falseHasPathWithoutReturn = hasPathWithoutReturn(node.vFalseBody);
+
+    if (falseHasPathWithoutReturn) {
+      for (const liveVar of liveVarsBefore) {
+        const child = createCairoTempVar(liveVar, ast);
+        node.vFalseBody.appendChild(child);
+        ast.registerChild(child, node.vFalseBody);
+      }
```

Comment:
The for loop could be extracted into a separate method `setBlockTempVars(ast, liveVars, block)` and used both by true and false bodies